azure-devops/run_build.yml

```diff
@@ -23,22 +23,26 @@ jobs:
         filePath: $(Build.SourcesDirectory)/azure-devops/install_msvc_preview.ps1
     - task: CacheBeta@0
       displayName: Cache vcpkg
+      timeoutInMinutes: 5
       inputs:
         key: $(vcpkgResponseFile) | $(Build.SourcesDirectory)/.git/modules/vcpkg/HEAD
         path: '$(vcpkgLocation)'
     - task: run-vcpkg@0
       displayName: 'Run vcpkg to Install boost-build'
+      timeoutInMinutes: 5
       inputs:
         vcpkgArguments: 'boost-build:x86-windows'
         vcpkgDirectory: '$(vcpkgLocation)'
     - task: run-vcpkg@0
       displayName: 'Run vcpkg'
+      timeoutInMinutes: 5

```

Comment:
5min may not always be enough here?

Reply:
esp. given we are caching this could lead to flakiness

Reply:
I will rethink the timeouts.

Reply:
New strategy:
* Added a 30 minute timeout for "Install MSVC Preview". This is currently skipped, but if it runs, a half-hour should be plenty.
* I changed all vcpkg timeouts to 10 minutes. These steps usually take 50s, 1s, 1s, so 10 minutes seems extremely generous. (I don't understand why we need two run-vcpkg steps.)
* I decreased "Build Support Tools" to 2 minutes. This compiles and links 3 source files, which takes something like 8 seconds. If this starts taking 10x more time, we should be informed.
* I left clang-format at 5 minutes. It currently takes less than a minute with parallelization, but a completely serial run could take a couple of minutes. (I'd like to keep this one fairly low until we resolve the hanging issue.)
* I decreased "Validate Files" to 2 minutes. This takes less than a second (my program is quite fast), but I figured using 2 minutes for all "should be very fast" tasks was reasonable.
* I increased "Build the STL" to 10 minutes. Currently it takes something like 36 seconds, but we know this will increase as we add more flavors.

Reply:
yes, this is good by me.